README.md

```diff
@@ -1,15 +1,203 @@
+<details>
+  <summary><b>Table of Contents</b></summary>
+  <p>
 
-[^1]: `http-file` excluded due to design issue.
-[^2]: only guaranteed inside project's own code. unsafe can and is used by dependencies
+  - [🚀 Features](#features-)
+  - [📊 Minimum Supported Rust Version](#minimum-supported-rust-version-)
+  - [⚡ Quick Start](#quick-start-)
+  - [🗨️ FAQ (Frequently Asked Questions)](#faq-frequently-asked-questions-%EF%B8%8F)
+  - [📣 More Contributors Wanted](#more-contributors-wanted-)
+  - [💖 Supporting Xitca-web](#supporting-xitca-web-)
+  - [📘 Documentation](#documentation-)
+  - [🛣️ Roadmap](#roadmap-%EF%B8%8F)
+  - [🙋 Contributing](#contributing-)
+  - [📜 License](#license-)
+  - [🤝 Credits](#credits-)
+
+  </p>
+</details>
+
+# Features 🚀
+
+- 🎗️ Supports HTTP/1.x and HTTP/2
+- 💪 Powerful request routing with optional opt-in macros
+- 📼 Full Tokio compatibility
+- 📦 Cross crate integration with Tower.
+- 🔌 Client/server WebSockets support
+- 🗜️ Transparent content compression/decompression (br, gzip, deflate, zstd)
+- 🚰 Multipart streams
+- 📄 Static assets
+- 🔒 SSL support using Xitca-tls or rustls
+- 🔌 Middlewares (Logger, Tracing, etc)
+
+**[⬆️ Back to Top](#xitca-web)**
+
+# Minimum Supported Rust Version 📊
+
+The latest release of the crate supports 1.79 and above rust versions.
+
+**[⬆️ Back to Top](#xitca-web)**
+
+# Quick Start ⚡
+
+> For a list of other examples, see: [**Examples**](examples)
+
+To get started using Xitca-web add the following code to your `main.rs` file: 
+```rust
+use xitca_web::{handler::handler_service, middleware::Logger, route::get, App};
+
+async fn index() -> &'static str {
+    "Hello world!!"
+}
+
+#[tokio::main]
+async fn main() -> std::io::Result<()> {
+    App::new()
+        .at("/", get(handler_service(index)))
+        .enclosed(Logger::new())
+        .serve()
+        .bind("localhost:8080")?
+        .run()
+        .await
+}
+```
+
+And then add the following dependencies to your `Cargo.toml` file:
+
+```toml
+xitca-web = { version = "*" , features = ["logger"]}
+tokio = { version = "*", default-features = false, features = ["full"] }
+```
+
+And then run the application/server using `cargo run` command and it should start the server/application on <http://127.0.0.1:8080>. Open your prefered browser and navigate to the address on which the server has started and you should see a `hello world!!` message appear.
+
+> [!Warning]
+> The project is still not stable. So expect breaking changes to appear at any time.
+
+**[⬆️ Back to Top](#xitca-web)**
+
+# FAQ (Frequently Asked Questions) 🗨️
+
+## Why Yet Another HTTP Framework?
+
+The primary reason behind creating Xitca-web is to have more memory safe HTTP framework which uses 100% safe rust unlike HTTP frameworks like actix-web which heavily rely upon unsafe rust code. As well as to provide a framework with low-memory footprint and with very small dependency tree.
```

Comment:
We should not mention any specific Rust crate in way that can possibly lead to negativity and contention. I feel we can just state xitca-web is 100% safe and remove comparison between actix-web.

Reply:
> We should not mention any specific Rust crate in way that can possibly lead to negativity and contention. I feel we can just state xitca-web is 100% safe and remove comparison between actix-web.

I don't think we are making a direct comparison here like if you read through it I am mentioning more like in general comparison to other projects and if you read carefully you will find that `actix-web` is just used as kind of an example. :sweat_smile: . But if you still suggest having it removed then I would not mind doing so. :slightly_smiling_face:

Reply:
It's not really about comparison but the possibility of misinterpretation and misunderstanding. For example a reader may form a negative view towards the project from your statement for following reason:
1. `actix-web` does not rely heavily on unsafe. It's code base has less than a dozen direct usage of `unsafe` key word. That's a very low amount of usage judging by it's code base.
2. `actix-web` has it's unsafe usage audited and there is no apparent soundness issue.
3. `usafe` is a language feature. Using it or not is based on design choice, trade off and execution. In most use cases it's an opinion of people and there could be no objectively correct answer (if unsafe should be used or not)

In other words it's very hard to make fair comparison and word it correctly in a short readme. And the best thing to do is avoid making comparison when possible. Library users are smart enough to figure things out themselves and form their own opinion.

Reply:
> It's not really about comparison but the possibility of misinterpretation and misunderstanding. For example a reader may form a negative view towards the project from your statement for following reason:
> 
> 1. `actix-web` does not rely heavily on unsafe. It's code base has less than a dozen direct usage of `unsafe` key word. That's a very low amount of usage judging by it's code base.
> 2. `actix-web` has it's unsafe usage audited and there is no apparent soundness issue.
> 3. `usafe` is a language feature. Using it or not is based on design choice, trade off and execution. In most use cases it's an opinion of people and there could be no objectively correct answer (if unsafe should be used or not)
> 
> In other words it's very hard to make fair comparison and word it correctly in a short readme. And the best thing to do is avoid making comparison when possible. Library users are smart enough to figure things out themselves and form their own opinion.

Ah ok I see, you are right. Ok then I will just reframe this section to not compare other projects in anyway. Thanks :heart: for your time explaining everything in depth. :slightly_smiling_face: